test/e2e/utils/olm.go

```diff
@@ -0,0 +1,121 @@
+package utils
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/infinispan/infinispan-operator/pkg/kubernetes"
+	"github.com/operator-framework/api/pkg/manifests"
+	coreos "github.com/operator-framework/api/pkg/operators/v1alpha1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+func init() {
+	ExpectNoError(coreos.AddToScheme(Scheme))
+}
+
+func (k TestKubernetes) CreateSubscription(sub *coreos.Subscription) {
+	err := k.Kubernetes.Client.Create(context.TODO(), sub)
+	ExpectNoError(err)
+}
+
+// TODO change name
+func (k TestKubernetes) Subscription(sub *coreos.Subscription) {
+	err := k.Kubernetes.Client.Get(context.TODO(), types.NamespacedName{Name: sub.Name, Namespace: sub.Namespace}, sub)
+	ExpectNoError(err)
+}
+
+// TODO rework to use provided Subscription
+func (k TestKubernetes) UpdateSubscriptionChannel(newChannel string, sub *coreos.Subscription) {
+	retryOnConflict(func() error {
+		k.Subscription(sub)
+		sub.Spec.Channel = newChannel
+		return k.Kubernetes.Client.Update(context.TODO(), sub)
+	})
+}
+
+func (k TestKubernetes) InstallPlan(sub *coreos.Subscription) *coreos.InstallPlan {
+	ipRef := sub.Status.InstallPlanRef
+	if ipRef == nil {
+		return nil
+	}
+	installPlan := &coreos.InstallPlan{}
+
+	err := k.Kubernetes.Client.Get(context.TODO(), types.NamespacedName{Name: ipRef.Name, Namespace: ipRef.Namespace}, installPlan)
+	ExpectNoError(err)
+	return installPlan
+}
+
+func (k TestKubernetes) ApproveInstallPlan(sub *coreos.Subscription) {
+	retryOnConflict(func() error {
+		installPlan := k.InstallPlan(sub)
+		installPlan.Spec.Approved = true
+		return k.Kubernetes.Client.Update(context.TODO(), installPlan)
+	})
+}
+
+func (k TestKubernetes) InstalledCSV(sub *coreos.Subscription) *coreos.ClusterServiceVersion {
+	csvName := sub.Status.InstalledCSV
+	csv := &coreos.ClusterServiceVersion{}
+	err := k.Kubernetes.Client.Get(context.TODO(), types.NamespacedName{Name: csvName, Namespace: sub.Namespace}, csv)
+	ExpectNoError(err)
+	return csv
+}
+
+func (k TestKubernetes) InstalledCSVServerImage(sub *coreos.Subscription) string {
+	csv := k.InstalledCSV(sub)
+	envName := "RELATED_IMAGE_OPENJDK"
+	envVars := csv.Spec.InstallStrategy.StrategySpec.DeploymentSpecs[0].Spec.Template.Spec.Containers[0].Env
+	index := kubernetes.GetEnvVarIndex(envName, &envVars)
+	if index < 0 {
+		panic(fmt.Sprintf("Unable to find env var '%s' in CSV %s", envName, csv.Name))
+	}
+	return envVars[index].Value
+}
+
+func (k TestKubernetes) PackageManifest(name, namespace string) *manifests.PackageManifest {
```

Comment:
This should also filter out the result based on CatalogSourceName. If both `test-catalog` and `released` catalog exist in same namespace then picking up one for parsing is random

Reply:
I'm not sure this is possible via the [REST API](https://docs.openshift.com/container-platform/4.8/rest_api/operatorhub_apis/packagemanifest-packages-operators-coreos-com-v1.html#apispackages-operators-coreos-comv1namespacesnamespacepackagemanifestsname). 

If two `CatalogSources` exist in the same namespace with entry "infinispan" then the most recently created is always returned. This is the case when executing a GET to `/apis/packages.operators.coreos.com/v1/namespaces/{namespace}/packagemanifests/{name}` and executing `kubectl get PackageManifest {name} -n {namespace}`

The entire behaviour of `PackageManifest` is strange with it not being a traditional k8s resource.

Reply:
It should be possible to retrieve all PackageManifests and filter out the one we want. But it's not a must have. Can be workaroundable by disabling default CatalogSources or creating the test CatalogSource in different namespace

Reply:
I have been unable to find a way todo this so far. Ideally I don't want to spend any more time on this atm, but it's something that we can come back to in the future.

Reply:
I've changed it so that we get all manifests via `/apis/packages.operators.coreos.com/v1/packagemanifests` and filter on the catalog name.